From e9f1528681098606bf92e95a78d25cfd53104e6c Mon Sep 17 00:00:00 2001
From: yuanjiahsu <yuanjiahsu@google.com>
Date: Thu, 6 Sep 2018 19:54:15 +0800
Subject: [PATCH 36/70] Use feature flag in Settings.

User can enable/disable faster emergency phone call feature by switching
feature flag from Settings, and the old flag switched by adb command was
replaced by this feature flag.

Test: Manually

Bug: 112168722
Bug: 113539598

Change-Id: Iaa2389845301d387da5924daa368169a8e6768c1
Merged-In: Iaa2389845301d387da5924daa368169a8e6768c1
---
 src/com/android/phone/EmergencyDialer.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/com/android/phone/EmergencyDialer.java b/src/com/android/phone/EmergencyDialer.java
index 11da00344..b9a64719e 100644
--- a/src/com/android/phone/EmergencyDialer.java
+++ b/src/com/android/phone/EmergencyDialer.java
@@ -62,6 +62,7 @@ import android.text.TextUtils;
 import android.text.TextWatcher;
 import android.text.method.DialerKeyListener;
 import android.text.style.TtsSpan;
+import android.util.FeatureFlagUtils;
 import android.util.Log;
 import android.util.TypedValue;
 import android.view.HapticFeedbackConstants;
@@ -329,8 +330,8 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
 
         getWindow().setAttributes(lp);
 
-        mAreEmergencyDialerShortcutsEnabled = Settings.Global.getInt(getContentResolver(),
-                Settings.Global.FASTER_EMERGENCY_PHONE_CALL_ENABLED, 0) != 0;
+        mAreEmergencyDialerShortcutsEnabled = FeatureFlagUtils
+                .isEnabled(this, FeatureFlagUtils.EMERGENCY_DIAL_SHORTCUTS);
         Log.d(LOG_TAG, "Enable emergency dialer shortcut: "
                 + mAreEmergencyDialerShortcutsEnabled);
 
-- 
2.17.1

